c10/core/DispatchKeySet.h

```diff
@@ -120,6 +120,44 @@ class DispatchKeySet final {
 private:
   DispatchKeySet(uint64_t repr) : repr_(repr) {}
   uint64_t repr_ = 0;
+
+public:
+  // STL iterator for DispatchKeySet. Iterates from the first DispatchKey to the
+  // NumDispatchKey enum. The iterate is only invalidated by the destruction of
+  // the underlying DispatchKeySet as the iterator stores a pointer to the raw
+  // represenation of the DispatchKeySet.
```

Comment:
AKA `std::list` style iterators, amirite? (Holding a pointer is pretty interesting, but it makes the iterators update if the input mutated so I guess that's a reasonable reason to do it that way.)

Reply:
Ya that's correct, it's a list style iterator. None of the functions in `DispatchKeySet` actually mutate the set; instead, they return a new set with mutation, so the input representation shouldn't ever mutate.